src/python_bindings.cpp

```diff
@@ -0,0 +1,196 @@
+#include <memory>
+#include <stdexcept>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/gil.h>
+#include <pybind11/eval.h>
+#include <pybind11/chrono.h>
+#include <pybind11/pytypes.h>
+#include <pybind11/stl.h>
+
+#include "behaviortree_cpp/basic_types.h"
+#include "behaviortree_cpp/bt_factory.h"
+#include "behaviortree_cpp/action_node.h"
+#include "behaviortree_cpp/tree_node.h"
+
+namespace BT
+{
+
+namespace py = pybind11;
+
+class Py_SyncActionNode : public SyncActionNode
+{
+public:
+  Py_SyncActionNode(const std::string& name, const NodeConfig& config) :
+    SyncActionNode(name, config)
+  {}
+
+  NodeStatus tick() override
+  {
+    PYBIND11_OVERRIDE_PURE_NAME(NodeStatus, Py_SyncActionNode, "tick", tick);
+  }
+};
+
+class Py_StatefulActionNode final : public StatefulActionNode
+{
+public:
+  Py_StatefulActionNode(const std::string& name, const NodeConfig& config) :
+    StatefulActionNode(name, config)
+  {}
+
+  NodeStatus onStart() override
+  {
+    PYBIND11_OVERRIDE_PURE_NAME(NodeStatus, Py_StatefulActionNode, "on_start", onStart);
+  }
+
+  NodeStatus onRunning() override
+  {
+    PYBIND11_OVERRIDE_PURE_NAME(NodeStatus, Py_StatefulActionNode, "on_running",
+                                onRunning);
+  }
+
+  void onHalted() override
+  {
+    PYBIND11_OVERRIDE_PURE_NAME(void, Py_StatefulActionNode, "on_running", onRunning);
```

Comment:
shouldn't this be
```suggestion
    PYBIND11_OVERRIDE_PURE_NAME(void, Py_StatefulActionNode, "on_halted", onHalted);
```
?

Reply:
Thanks! Fixed.